src/lib.rs

```diff
@@ -441,6 +444,73 @@ impl versioned_store_permissions::CredentialChecker<Runtime> for SudoKeyHasAllCr
         <sudo::Module<Runtime>>::key() == *account
     }
 }
+
+parameter_types! {
+    pub const CurrentLeadCredential: Credential = 0;
+    pub const AnyActiveCuratorCredential: Credential = 1;
+    pub const AnyActiveChannelOwnerCredential: Credential = 2;
+    pub const PrincipalIdMappingStartsAtCredential: Credential = 1000;
+}
+
+pub struct ContentWorkingGroupCredentials {}
+impl versioned_store_permissions::CredentialChecker<Runtime> for ContentWorkingGroupCredentials {
```

Comment:
Why is this implemented here? There already is this here

https://github.com/mnaamani/substrate-runtime-joystream/blob/integrate-content-wg/src/content_working_group/lib.rs#L2017

I do not recall how to think about this mapping process, but hte fact that there is iteration in this implementaiton makes me think this is almost certainly wrong.

Reply:
That implementation (in the working group itself) only does one to one mapping. I needed to support a range of Credentials that represented a group, so I made it as a separate impl, which re-uuses the impl you linked for range of credentials that represent 1:1 mapping to a principal.

Reply:
Yes the need for iteration is because there is a single map of principalId -> principal, so it was tge only way to identify different principal types (without adding more state)